0003-I-dapr-compose.md

```diff
@@ -0,0 +1,294 @@
+
+![Foreground process](./resources/0003-I-dapr-compose/dapr-compose-flow-2.png)
+
+This way as long as `dapr compose` _process_ is also running, the state of all the processes that it starts can be stored in memory and this process itself can monitor a process if it crashes and restarts as needed.
+
+The second approach is what is proposed as an initial implementation in **v1.10**. Later, this can be expanded to having a detached mode and a _daemon service_ that monitors the applications as shown in the first approach.
+
+### Logging options
+
+Right `dapr run` executes as a foreground interactive process, and both the `daprd` logs and associated application logs are directly written to the STDOUT of the `dapr run` _process shell_ and it is not stored anywhere.
+
+Considering that executing `dapr compose` will be run multiple applciations, routing all the logs for all applications and `daprd` processes will make the STDOUT completely chaotic and the user will be overwhelmed with log output.
+
+For example, consider two applications `order-proc` and `checkout` that are run on executing `dapr compose` and logs are routed to STDOUT:
+
+```
+<order-proc> ==APP== waiting for daprd to start
+<order-proc> INFO[0000] enabled gRPC tracing middleware               app_id=order-proc instance=Mukundans-MacBook-Pro.local scope=dapr.runtime.grpc.api type=log ver=1.9.3
+<checkout> ==APP==pinging dapr API
+<checkout> INFO[0000] enabled gRPC tracing middleware               app_id=checkout instance=Mukundans-MacBook-Pro.local scope=dapr.runtime.grpc.api type=log ver=1.9.3
+<order-proc> INFO[0000] started daprd              app_id=order-proc instance=Mukundans-MacBook-Pro.local scope=dapr.runtime.grpc.api type=log ver=1.9.3
+<order-proc> ==APP== starting the application
+<order-proc> ==APP== processing request 1
+<order-proc> ==APP== processing request 2
+<checkout> INFO[0000] started daprd               app_id=checkout instance=Mukundans-MacBook-Pro.local scope=dapr.runtime.grpc.api type=log ver=1.9.3
+<checkout> ==APP== processing request 2
+<order-proc> ==APP== processing request 3. request 3 calls dapr API
+<order-proc> INFO[0000] requst 3 is calls dapr API             app_id=order-proc instance=Mukundans-MacBook-Pro.local scope=dapr.runtime.grpc.api type=log ver=1.9.3
+<checkout> ==APP== processing request 3
+
+```
+
+The logs will be as shown above, it will be chaotic to see what application is in which state and so on.
+
+Instead of writing to STDOUT, each application and associated `daprd` process will write the logs to `{application dir}/.dapr/logs/{app id}/app_{datetime}.log`, `{application dir}/.dapr/logs/{app id}/daprd_{datetime}.log`.
+
+### Subcommands and flags for `compose` command in CLI
+
+It is proposed to have a subcommand `up` in `compose` command to _bring up_ all the applications and daprd sidecars. Later on an extension can be added to _shutdown all_ application and sidecars with a `down` subcommand.
```

Comment:
Why introduce the 'up' command and simply have just `compose` to start all the applications?

Reply:
This was mostly following the same way `docker compose`, it will be familiar to the users.